### PR TITLE
feat(call-service): implement inbound call state machine (IN-3.3)

### PR DESCRIPTION
**Pull request by pcamarajr**
## Summary

This PR implements the call state machine for inbound calls (IN-3.3), extending CallService with four methods to handle inbound call state transitions: answer, reject, cancel, and BYE.

## Changes

- **handle_inbound_answer()**: Transitions inbound call from Ringing → Active state, sets start_time, and emits state change event
- **handle_inbound_reject()**: Transitions inbound call from Ringing → Ended state, sets end_time, and emits state change event
- **handle_inbound_cancel()**: Handles CANCEL requests for inbound calls in Ringing state, transitions to Ended
- **handle_inbound_bye()**: Handles BYE requests for active inbound calls, stops RTP session, and transitions to Ended
- Added comprehensive unit tests for all four methods (20 tests total)
- Updated documentation to mark IN-3.3 as completed

### How to Test

1. **Setup**:
   ```bash
   git checkout feat/inbound-call-state-machine
   cd src-tauri
   cargo test --lib call_service::tests
   ```

2. **Test Coverage**:
   - ✅ All 20 unit tests passing
   - ✅ Success cases for all four methods
   - ✅ Error cases: invalid state, wrong direction, call not found
   - ✅ State transition validation
   - ✅ Event emission verification

3. **Expected Behavior**:
   - All methods validate call direction (must be inbound)
   - All methods validate call state (answer/reject/cancel require Ringing, BYE requires Active)
   - State transitions follow the state machine rules
   - Events are emitted on state changes
   - RTP session is stopped when handling BYE

### Test Coverage

- [x] Unit tests added/updated (20 tests)
- [x] Integration tests (not applicable - these are service methods)
- [x] Manual testing completed (via unit tests)
- [x] Edge cases tested (invalid states, wrong direction, call not found)

### Code Quality

- [x] Code follows project conventions
- [x] Functions are focused and single-purpose
- [x] Variable names are descriptive
- [x] No code duplication
- [x] Error handling is appropriate

### Framework Compliance

- [x] Rust code follows idiomatic patterns
- [x] Tauri IPC follows best practices (methods ready for command integration)
- [x] Design system components used correctly (N/A - backend only)

### Security

- [x] No hardcoded secrets
- [x] Input validation present (call direction and state validation)
- [x] Sensitive data handled securely
- [x] Tauri capabilities properly configured (N/A - no new commands yet)

### Documentation

- [x] README.md updated (marked IN-3.3 as completed)
- [x] Code comments explain complex logic
- [x] Architecture docs updated (roadmap marked IN-3.3 as completed)

## Breaking Changes

None - This is a new feature addition that extends existing functionality.

## Related Issues

Implements IN-3.3 from the implementation roadmap.

## Additional Notes

- All methods follow the same pattern as existing outbound call handlers
- State validation ensures calls can only transition through valid states
- Event emission ensures frontend stays synchronized with backend state
- RTP session cleanup is handled in handle_inbound_bye() to prevent resource leaks
- Methods are ready for integration with Tauri commands in future tasks (IN-3.4, IN-3.5)

### CODE REVIEW

File: src-tauri/src/services/call_service.rs [hunk 1]
```diff
@@ -954,6 +954,318 @@ impl CallService {
...
+    /// # Returns
+    /// `Ok(())` if call was ended successfully, `Err(SipError)` otherwise
+    pub async fn handle_inbound_bye(&self, call_id: &CallId) -> Result<(), SipError> {
+        eprintln!(
+            "DEBUG:[CALL_SERVICE/INBOUND_BYE] Handling BYE for inbound call: {}",
+            call_id.as_str()
+        );
+
+        // Stop RTP session if active
+        self.stop_rtp_session(call_id).await;
```
Comment (pcamarajr) [inline src-tauri/src/services/call_service.rs]: **💡 Suggestion:** Consider validating call existence before stopping RTP session

While this is safe (since `stop_rtp_session()` handles missing sessions gracefully) and matches the pattern in `end_call()`, validating the call exists first would provide clearer error semantics. If the call doesn't exist, we'd return an error immediately rather than attempting to stop a non-existent RTP session first.

This is optional and not a blocker - the current implementation is correct and consistent with existing code.

Comment (pcamarajr) [reply src-tauri/src/services/call_service.rs]: ✅ Resolved in commit cc882a51f648027c07a7df85bc1e79e829a248da

Moved call validation before RTP session stop to provide clearer error semantics. If the call doesn't exist, we now return an error immediately rather than attempting to stop a non-existent RTP session first.
